tests/validators/test_dataclasses.py

```diff
@@ -584,3 +584,180 @@ def validate_b(
     v = SchemaValidator(schema)
     foo = v.validate_python({'a': 1, 'b': b'hello'})
     assert dataclasses.asdict(foo) == {'a': 1, 'b': 'hello world!'}
+
+
+def test_dataclass_self_init():
+    @dataclasses.dataclass(init=False)
+    class Foo:
+        a: str
+        b: bool
+
+        def __init__(self, *args, **kwargs):
+            v.validate_python(ArgsKwargs(args, kwargs), self_instance=self)
```

Comment:
It might be nice to use `__init_subclass__` to do `cls.v = SchemaValidator(...)`. As is the test looks a bit confusing and it might be hard to connect it with how it's used in `pydantic`.

Reply:
this is pretty much exactly what we do in pydantic, should work fine.